wallet/src/account_manager/mod.rs

```diff
@@ -124,32 +125,32 @@ impl AccountManager {
 
     /// Get the balance of all accounts added together
     pub async fn balance(&self) -> crate::Result<AccountBalance> {
+        let mut balances = Vec::new();
         let accounts = self.accounts.read().await;
 
-        let mut balances = Vec::new();
+        #[allow(clippy::significant_drop_in_scrutinee)]
         for account in accounts.iter() {
-            let account_balance = account.balance().await?;
-            balances.push(account_balance);
+            balances.push(account.balance().await?);
         }
 
-        let balance = add_balances(balances)?;
+        drop(accounts);
 
-        Ok(balance)
+        add_balances(balances)
     }
 
     /// Sync all accounts
     pub async fn sync(&self, options: Option<SyncOptions>) -> crate::Result<AccountBalance> {
+        let mut balances = Vec::new();
         let accounts = self.accounts.read().await;
 
-        let mut balances = Vec::new();
+        #[allow(clippy::significant_drop_in_scrutinee)]
         for account in accounts.iter() {
```

Comment:
Why not also
```suggestion
        for account in self.accounts.read().await.iter() {
```
here?